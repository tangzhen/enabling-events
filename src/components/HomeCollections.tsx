import { Paper, Typography } from "@material-ui/core";
import React from "react";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";
import SendIcon from "@material-ui/icons/Send";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    container: {
      color: theme.palette.common.white,
      position: "relative",
      marginTop: theme.spacing(4),
      marginBottom: theme.spacing(4),
      backgroundSize: "cover",
      backgroundRepeat: "no-repeat",
      backgroundPosition: "center",
      backgroundImage: "url('http://localhost:3001/collection-1.jpg')",
    },
    content: {
      padding: theme.spacing(10),
    },
    mask: {
      background:
        "linear-gradient(90.17deg, rgba(0, 125, 129, 0.8) 10.73%, rgba(0, 117, 120, 0) 99.86%)",
    },
    sendButton: {
      marginTop: theme.spacing(4),
      color: theme.palette.primary.main,
      borderRadius: "20px",
    },
  })
);

export default function HomeCollections() {
  const classes = useStyles();

  return (
    <Paper className={classes.container}>
      <Grid container className={classes.mask}>
        <Grid item xs={12} sm={6}>
          <div className={classes.content}>
            <Typography variant="h3" gutterBottom>
              Enabling Events
            </Typography>
            <Typography gutterBottom>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit ut
              aliquam, purus sit amet luctus venenatis, lectus magna fringilla
              urna, porttitor rhoncus dolor purus non enim praesent elementum
              facilisis leo, vel fringilla est ullamcorper eget nulla facilisi
              etiam dignissim diam quis enim lobortis scelerisque fermentum dui
              faucibus in ornare quam viverra orci sagittis eu volutpat odio
              facilisis mauris sit amet massa vitae tortor condimentum.
            </Typography>
            <Button
              variant="contained"
              endIcon={<SendIcon />}
              className={classes.sendButton}
            >
              BROWSE EVENTS
            </Button>
          </div>
        </Grid>
      </Grid>
    </Paper>
  );
}
