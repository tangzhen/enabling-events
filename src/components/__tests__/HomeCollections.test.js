import React from "react";
import { render } from "../../../test/test-utils";
import HomeCollections from "../HomeCollections";

describe("HomeCollections", () => {
  it("should render component correct", () => {
    const { container } = render(<HomeCollections />);
    expect(container).toMatchSnapshot();
  });

  it("should render the browser events button in the collections view", () => {
    const { getByText } = render(<HomeCollections />);
    const browseEventsElement = getByText(/BROWSE EVENTS/i);
    expect(browseEventsElement).toBeInTheDocument();
  });
});
