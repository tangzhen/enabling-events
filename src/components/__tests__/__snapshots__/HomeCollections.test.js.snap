// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCollections should render component correct 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root makeStyles-mask-3 MuiGrid-container"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="makeStyles-content-2"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
          >
            Enabling Events
          </h3>
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis, lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo, vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis enim lobortis scelerisque fermentum dui faucibus in ornare quam viverra orci sagittis eu volutpat odio facilisis mauris sit amet massa vitae tortor condimentum.
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-sendButton-4"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              BROWSE EVENTS
              <span
                class="MuiButton-endIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
                  />
                </svg>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
